File: CS/GrowthWare.Framework/Model/Profiles/Interfaces/IMGroupRolePermissionSecurity.cs
using System;
using System.Collections.Generic;
using System.Collections.ObjectModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace GrowthWare.Framework.Model.Profiles.Interfaces
{
    /// <summary>
    /// Ensures the basic properties are avalible to all Profile model objects.
    /// </summary>
    /// <remarks>
    /// If it is decided to use entities in the future then
    /// this interface should be used for the save, delete, and getitem methods.
    ///  </remarks>
    [CLSCompliant(true)]
    public interface IMGroupRolePermissionSecurity
    {
        Collection<string> AddGroups
        {
            get;
        }
        Collection<string> DeleteGroups
        {
            get;
        }
        Collection<string> EditGroups
        {
            get;
        }
        Collection<string> ViewGroups
        {
            get;
        }
         Collection<string> AssignedAddRoles
        {
            get;
        }
        Collection<string> AssignedDeleteRoles
        {
            get;
        }
        Collection<string> AssignedEditRoles
        {
            get;
        }
        Collection<string> AssignedViewRoles
        {
            get;
        }
        Collection<string> DerivedAddRoles
        {
            get;
        }
        Collection<string> DerivedDeleteRoles
        {
            get;
        }
        Collection<string> DerivedEditRoles
        {
            get;
        }
        Collection<string> DerivedViewRoles
        {
            get;
        }
    }
}

File: VB/GrowthWare.Framework/Model/Profiles/Interfaces/IMGroupRolePermissionSecurity.vb
Imports System.Collections.ObjectModel

Namespace Model.Profiles.Interfaces
    <CLSCompliant(True)> _
    Public Interface IMGroupRolePermissionSecurity

        ''' <summary>
        ''' Add groups that are directly assigned
        ''' </summary>
        ReadOnly Property AddGroups As Collection(Of String)

        ''' <summary>
        ''' Edit groups that are directly assigned
        ''' </summary>
        ReadOnly Property EditGroups As Collection(Of String)

        ''' <summary>
        ''' Delete groups that are directly assigned
        ''' </summary>
        ReadOnly Property DeleteGroups As Collection(Of String)

        ''' <summary>
        ''' View groups that are directly assigned
        ''' </summary>
        ReadOnly Property ViewGroups As Collection(Of String)

        ''' <summary>
        ''' Add roles that are directly assigned
        ''' </summary>
        ReadOnly Property AssignedAddRoles As Collection(Of String)

        ''' <summary>
        ''' Edit roles that are directly assigned
        ''' </summary>
        ReadOnly Property AssignedEditRoles As Collection(Of String)

        ''' <summary>
        ''' Delete roles that are directly assigned
        ''' </summary>
        ReadOnly Property AssignedDeleteRoles As Collection(Of String)

        ''' <summary>
        ''' View roles that are directly assigned
        ''' </summary>
        ReadOnly Property AssignedViewRoles As Collection(Of String)

        ''' <summary>
        ''' Add roles that are derived from roles assigned to groups
        ''' </summary>
        ReadOnly Property DerivedAddRoles As Collection(Of String)

        ''' <summary>
        ''' Edit roles that are derived from roles assigned to groups
        ''' </summary>
        ReadOnly Property DerivedEditRoles As Collection(Of String)

        ''' <summary>
        ''' Delete roles that are derived from roles assigned to groups
        ''' </summary>
        ReadOnly Property DerivedDeleteRoles As Collection(Of String)

        ''' <summary>
        ''' View roles that are derived from roles assigned to groups
        ''' </summary>
        ReadOnly Property DerivedViewRoles As Collection(Of String)
    End Interface
End Namespace
